### PR TITLE
refs #63 API to get events by date

### PR DESCRIPTION
**Pull request by maheshkurma**
Build the data from the agencies >events > event_dates, call to build out "Events Nearby Today"

example URL - localhost:8888/agencies?zip_code=34219&event_date_on=2020-10-08

### CODE REVIEW

File: app/controllers/api/agencies_controller.rb [hunk 1]
```diff
@@ -3,17 +3,17 @@
...
     def index
-      if (@zip = search_params[:zip_code])
-        @agencies = @agencies.by_zip_code(@zip)
-      end
-      if (date = search_params[:event_date])
-        @agencies = @agencies.with_event_after(date.delete('-'))
-      end
+      by_zip(@zip)
+
+      by_event_date_after(@event_date)
```
Comment (PhilNorman2) [inline app/controllers/api/agencies_controller.rb:14]: I think 'by_event_date_on_or_after' would be more descriptive than 'by_event_date_after'.   I realize you are following the previous model method naming but if we ever truly need event_date_after we would have to make this change.  At least something to keep in mind.

File: app/models/agency.rb [hunk 2]
```diff
@@ -21,6 +21,11 @@ class Agency < ApplicationRecord
 
   scope :active, -> { where(status_id: 1) }
 
+  scope :with_event_on, lambda { |date|
+    joins(:event_dates)
+      .where('event_dates.event_date_key = ?', date)
+  }
+
   scope :with_event_after, lambda { |date|
```
Comment (PhilNorman2) [inline app/models/agency.rb:29]: similar comment on scope name.  Could change to 'with_event_on_or_after'

File: spec/models/agency_spec.rb [hunk 3]
```diff
@@ -80,5 +80,22 @@
...
+        date = (Date.today + i).to_s.delete('-')
+        agency = create(:agency)
+        event = create(:event, agency: agency)
+        create(:event_date, event: event, date: date)
+        agency
+      end
+
+      target_date = Date.today.to_s.delete('-')
+
+      expected_agency_ids = agencies[0..0].pluck(:id)
```
Comment (PhilNorman2) [inline spec/models/agency_spec.rb:95]: I would also like to see a negative test for with_event_on.  For example a test where the expected_agency_id does not match target_date.

**Review by PhilNorman2 — COMMENTED** [review]
Just a few minor comments.

File: app/models/agency.rb (hunk 2 above)
Comment (maheshkurma) [reply app/models/agency.rb:29]: here this scope pulls only events on date (ex: 2020-10-10) so i just added name as `with_event_on`

File: app/controllers/api/agencies_controller.rb (hunk 1 above)
Comment (maheshkurma) [reply app/controllers/api/agencies_controller.rb]: updated

File: spec/models/agency_spec.rb (hunk 3 above)
Comment (maheshkurma) [reply spec/models/agency_spec.rb]: yeah @PhilNorman2  i agree it would be nice to cover negative test too... thanks updated.

**Review by PhilNorman2 — APPROVED** [review]
Approved.

**Comment by sarath-yerra** [comment]
We can close this branch, In the Freshtrak client i am using existing EVENTS_LIST API.No need to make another api call.

**Comment by sarath-yerra** [comment]
EVENTS_LIST gets events for next 30 days.
I am filtering Todays events and weekly events using EVENTS_LIST

**Pull request closed without merging**